pkg/operation/seed/components.go

```diff
@@ -83,8 +83,8 @@ func defaultEtcdDruid(
 	return etcd.NewBootstrapper(c, v1beta1constants.GardenNamespace, conf, image.String(), imageVectorOverwrite), nil
 }
 
-func defaultKubeStateMetrics(c client.Client, imageVector imagevector.ImageVector) (component.DeployWaiter, error) {
-	image, err := imageVector.FindImage(images.ImageNameKubeStateMetrics)
+func defaultKubeStateMetrics(c client.Client, imageVector imagevector.ImageVector, seedVersion string) (component.DeployWaiter, error) {
+	image, err := imageVector.FindImage(images.ImageNameKubeStateMetrics, imagevector.TargetVersion(seedVersion))
```

Comment:
Can you make sure the image vector constraints work with a version suffix like the one used by GKE?
e.g. something like `v1.20.0-gke.300`

Reply:
Yes, it will work.